packages/headless/doc-parser/src/actions-resolver.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  ApiEntryPoint,
+  ApiItemKind,
+  ApiVariable,
+} from '@microsoft/api-extractor-model';
+import {DocNode, DocParamBlock} from '@microsoft/tsdoc';
+import {AnyEntity, Entity} from './entity';
+import {emitAsTsDoc} from './tsdoc-emitter';
+
+interface ActionsNamespace {
+  name: string;
+  members: Action[];
+}
+
+enum ActionKind {
+  Action = 'action',
+  AsyncAction = 'asyncThunk',
+  AnalyticsAction = 'analyticsAction',
+}
+
+interface Action {
+  desc: string;
+  fullName: string;
+  parameter: AnyEntity | null;
+  actionKind: ActionKind;
+}
+
+export function resolveActionNamespaces(
+  entry: ApiEntryPoint
+): ActionsNamespace[] {
+  return entry.members
+    .filter(
+      (member) =>
+        member.kind === ApiItemKind.Namespace &&
+        member.displayName.endsWith('Actions')
+    )
+    .map((namespace) => ({
+      name: namespace.displayName,
+      members: namespace.members
+        .filter((member) => member.kind === ApiItemKind.Variable)
+        .map((action) => resolveAction(action as ApiVariable))
+        .filter((action) => action) as Action[],
+    }));
+}
+
+function resolveAction(variable: ApiVariable): Action | null {
+  const tokens = variable.excerpt.tokens.map(({text}) => text);
+
+  if (tokens[1] === 'AsyncThunkDefinition') {
```

Comment:
For this PR, I only parsed async thunks since parsing other actions would take a lot more code.

Reply:
Interesting, I would have thought async thunks would be more complex to parse? Why are the sync actions more difficult?

Reply:
Oh I didn't mean more code relatively, I meant that it would take a bunch of extra code to parse normal actions and analytics actions, they would probably be easier to parse.